### PR TITLE
iptables: Address a TODO, check existing chain before chain deletion/creation

### PR DESCRIPTION
**Pull request by masap**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
iptables: Address a TODO, check existing chain before chain deletion/creation

The TODO was introduced by commit e045c6ce2fd81b85637894a1655a90bc367b90f7 "Split portals into host and container".

/cc @thockin

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:
```docs
NONE
```

### CODE REVIEW

**Comment by masap** [comment]
@thockin  Is there any comment ?

File: pkg/util/iptables/iptables.go [hunk 1]
```diff
@@ -246,13 +246,16 @@ func (runner *runner) EnsureChain(table Table, chain Chain) (bool, error) {
...
+	if err != nil {
+		return false, err
+	}
+	if exists {
+		return true, nil
+	}
+
 	out, err := runner.run(opCreateChain, fullArgs)
 	if err != nil {
-		if ee, ok := err.(utilexec.ExitError); ok {
```
Comment (thockin) [inline pkg/util/iptables/iptables.go:251]: I don't think we can remove this - it's always going to be a race

**Review by thockin — COMMENTED** [review]
Hi!  I appreciate the PR, but I don't think we should do this.  Here's why:

We trade one op (try create/delete) for best-case one op (check) and worst case two ops.  Unless we can show that the list op is significantly faster than the try-create/delete op, this is a net negative.  My (very quick!) testing shows that running a list op on an existing chain is about the same cost as just trying to create it. 

I think the right answer is just to remove the TODO and maybe add a comments that it's not worth checking explicitly because a "check" isn't any cheaper than a "do".

**Comment by masap** [comment]
> My (very quick!) testing shows that running a list op on an existing chain is about the same cost as just trying to create it.

Thank you for a comment.
Right. My PR is slower or even than previous for both ```EnsureChain``` and ```DeleteChain```.
These are averages of each 1000 times trials (unit is micro second).

||EnsureChain(create)|EnsureChain(not create)|DeleteChain(delete)|DeleteChain(not delete)|
|---|---|---|---|---|
|before|1897|1743|1783|1558|
|after|3735|1708|3627|1761|

> I think the right answer is just to remove the TODO and maybe add a comments
> that it's not worth checking explicitly because a "check" isn't any cheaper than a "do".

I fixed my commit as you said.

**Comment by masap** [comment]
@thockin Hi! How about this ?

**Pull request closed without merging**